sssd_test_framework/utils/authentication.py

```diff
@@ -324,18 +324,26 @@ def passkey(self, username: str, *, pin: str | int, device: str, ioctl: str, scr
         device_path = self.fs.upload_to_tmp(device, mode="a=r")
         ioctl_path = self.fs.upload_to_tmp(ioctl, mode="a=r")
         script_path = self.fs.upload_to_tmp(script, mode="a=r")
+        self.host.ssh.run(
+            "mv -f /usr/libexec/sssd/passkey_child /usr/libexec/sssd/passkey_child.orig", raise_on_error=False
```

Comment:
Hi,

you might run into issues if this is called multiple times in a test run because at the second call the script will be moved to `/usr/libexec/sssd/passkey_child.orig` which will then call itself recursively. Maybe it would be possible to use the backup/restore functionality for this, i.e. add `/usr/libexec/sssd/passkey_child` to the `backup()` and `restore()` calsl in `sssd_test_framework/hosts/client.py`. Then the `/usr/libexec/sssd/passkey_child` would be restored after each test. @pbrezina, is this a proper use of the backup/restore functionality or should the test do the cleanup itself, i.e. make sure `/usr/libexec/sssd/passkey_child.orig"` is moved back to `/usr/libexec/sssd/passkey_child`?

bye,
Sumit

Reply:
Hi,

thanks for the update, I'm fine with the change but please remove the old, commented-out part. @pbrezina, is it ok to use the backup/restore functionality this way?

bye,
Sumit

Reply:
Hello,

I am facing another issue, 
like if I backup and restore the from using methods form file `hosts/client.py` then tests failing, as I added only one thing is either truncate the `/usr/libexec/sssd/passkey_child` or completely remove it before entering in method  using `rm -f /usr/libexec/sssd/passkey_child` but still tests failed,
but If I add simple `mv -f` before and after the methods its working.

@pbrezina can you help here for appropriate approach, I tested this locally.

Reply:
Updated the code.

Reply:
Hi,

thanks, I'm fine with the latest version and will mark this conversation as resolved.

bye,
Sumit